### DISCUSSION TITLE
Rapid and Linear Speed

### DISCUSSION THREAD

**Discussion by Tribo-2dot0** [arkypita/LaserGRBL#1502] [General]
I still don't really understand the difference between Linear Speed and Rapid Speed... I thought the first one was for vector and Rapid for Raster, but I realized that Linear changes Raster...
Does Rapid change the speed of blank parts in Rasters?
Do they have another function in some other situation?
Thank you in advance for any response.

![Captura de Tela (74)](https://user-images.githubusercontent.com/83982525/123575385-fd0e1180-d7a7-11eb-949c-823325f3d5b6.png)

**Comment by arkypita** [comment]
Rapid = G0 non tracing fast movement
Linear = G1, G2, G3 tracing linear speed movement

**Comment by Tribo-2dot0** [comment]
Right... So when you're cutting, the cutting speed is G1 and when it moves between points it's G0... if I understand...

My doubt is when I'm recording raster image it behaves with two speeds, one for dark areas and another for light areas (where the laser turns off), in a B/W logo or even in a grayscale image where it also turns off the laser in the blank areas.

**Comment by arkypita** [comment]
> Right... So when you're cutting, the cutting speed is G1 

Is the F speed as requested, eventually limited by max speed configuration.

> and when it moves between points it's G0... if I understand.

It is the max speed configured.
